tests/php/core-api/test_class.jetpack-core-api-endpoint.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+class WP_Test_Jetpack_Core_Api_Endpoint extends WP_UnitTestCase {
+
+	public function setUp() {
+		parent::setUp();
+
+		require_once dirname( __FILE__ ) . '/../../../_inc/lib/core-api/class.jetpack-core-api-endpoint.php';
+	}
+
+	/**
+	 * @author zinigor
+	 * @covers Jetpack_Core_API_Endpoint
+	 * @requires PHP 5.2
+	 */
+	public function test_Jetpack_Core_API_Endpoint_class_structure() {
+
+		$reflection = new ReflectionClass( 'Jetpack_Core_API_Endpoint' );
+
+		$this->assertTrue( $reflection->isAbstract() );
+
+		$processMethod = $reflection->getMethod( 'process' );
+		$this->assertTrue( $processMethod->isPublic() );
+		$this->assertTrue( $processMethod->isAbstract() );
+	}
```

Comment:
I'm not sure how I feel about this kind of test. It's not really testing any behavior. Rather, it's merely confirming that the code properties are exactly as they are written in those classes.

_Why_ would we care if `process` is `public` and `abstract`? Is there any impact on the final product? Why not just look at that file and see them declared as such?

These kinds of tests risk providing little value while taking up maintenance overhead.